### DISCUSSION TITLE
MuSHR project in France

### DISCUSSION THREAD

**Discussion by maximericaud** [prl-mushr/mushr#104] [Show and tell]
#### Maxime Ricaud - graduate intern
<br>
<p align="center">
  <img src="https://user-images.githubusercontent.com/97990762/227251863-0ca62af7-a1a5-450c-aba9-29d1f9b80e73.jpeg" width="300">
</p>
<p align="center">
  <img src="https://user-images.githubusercontent.com/97990762/227252080-aa6eec84-267e-467c-a3c7-12702ba0a610.jpeg" alt= “First prototype” width="300">
</p>
We are at the continuity of the SlaSHR project currently the prototype have a non-final color and we will produce three more robots.

Our goal is to implement these robots to evaluate SLAM-VI RTAB-Map algorithms as suggested [here](https://github.com/prl-mushr/mushr/discussions/94#discussioncomment-4348239) and collaborative navigation algorithms like Halmitonian [[1]](#1).

We thank you for the tutorials provided, they are well structured and allowed us to quickly make the prototype functional.

Currently we have a small problem with the push button, indeed we have done a complete installation by hand and we have an error when launching the teleop [Issue #103](https://github.com/prl-mushr/mushr/issues/103).  
So we installed the image of the MuSHR provided [here](https://mushr.io/hardware/build_instructions/#sd-card-setup) we no longer have error messages however it remains non-functional.

We will present in a future post the four MuSHR as well as examples of implementation.

Finally we would like to know if a contribution for a tutorial on RTAB-Map would be possible.

<a id="1">[1]</a> C. Mavrogiannis and R. A. Knepper, “Hamiltonian coordination primitives for decentralized multiagent navigation,” The International Journal of Robotics Research, vol. 40, no. 10–11. SAGE Publications, pp. 1234–1254, Aug. 13, 2021. doi: 10.1177/02783649211037731.

**Comment by maximericaud** [comment]
<p align="center">
  <img src="https://github.com/prl-mushr/mushr/assets/97990762/a03dda07-4e46-4e20-bae4-57e314274f57" width="500"/>
</p>
<p align="center">
  <img src="https://github.com/prl-mushr/mushr/assets/97990762/83c6abc2-db0d-41ac-a604-785464341472" width="350"/>
</p>
Here is our platform composed of 4 MuSHR. Number 2 has a motor with encoder and number 3 has a RealSense T265 to explore the differences of SLAM.
<br/>
<br/>
We also have a small modular arena to carry out tests in real conditions on the SLAM and autonomous navigation. Here is an example of what we were able to achieve using RTAB-Map :

https://github.com/prl-mushr/mushr/assets/97990762/0312d0f8-4170-4cba-9e4e-46fd6fdf5c44

We were able to implement autonomous navigation on ROS Melodic both in simulation and physically.
For the multi-agent part we carried out the simulation by following and adapting the tutorial, do you think to complete the tutorial for a physical implementation?

Moreover, we have delved into the realm of multi-agent simulations, meticulously following and adapting tutorials to suit our specific goals. A question arises: could these simulations be extended to physical implementations? Your insights on this matter would be greatly appreciated.

We have also improved the performance of MuSHR by starting a migration to a Jetson Orin Nano. We will be sharing a comprehensive installation procedure in an upcoming post, providing a detailed guide for those interested in replicating our setup.

**Comment by sidtalia** [comment]
Hi Maxime, and hello to the rest of the team!

This looks amazing! We are very happy to see visual odometry actually being used on the MuSHR and would love to read any reports you have on this (we read the one from before, curious if there is a new one with any interesting findings).

As for the safety button; I think it may have been disabled from within the code, as there is already a "dead man's switch" on the controller to address this. 

As for contributing to the tutorials, yes, we should be able to let you contribute a tutorial! It is however recommended that you test that tutorial internally with undergrads who have little to no prior knowledge of robotics; the expectation from a tutorial is usually that the user should be able to follow through without hassle.

As for the Orin Nano migration; that sounds good! We are currently working on a docker that can work across all orin modules, so perhaps it would be good to wait and contribute to that instead. 

We're extremely happy to see these experiments and demos.

**Reply by sidtalia** [reply]
@maximericaud Hey, have you heard of this new [learned approach to odometry](https://github.com/uzh-rpg/learned_inertial_model_odometry) ? 
I would be curious as to how this compares to the visual approaches! The T265 should have a usable IMU, and you could use either MoCAP or the output of the lidar-particle filter to obtain ground truth for training.

The key advantage of a non-visual approach is that it can be trained on data obtained from either indoor or outdoor environments. We are currently working on a system that performs outdoor autonomy using GPS-IMU fusion, and we can obtain pose-twist labels relatively easily from that system.

**Reply by maximericaud** [reply]
Hello @sidtalia,
Thank you for this very interesting idea.
However, I am no longer currently an intern on this project, I will pass on your idea to my internship manager and to future students who will study it.
